src/workerd/io/actor-cache.c++

```diff
@@ -2047,6 +2068,14 @@ void ActorCache::putImpl(Lock& lock, kj::Own<Entry> newEntry,
           return;
         }
 
+        // Entry may have `countedDelete` indicating we're still waiting to get a count from a
```

Comment:
Huh, I'm pretty suspicous of this code movement -- there's nothing saying that a `PRESENT` value can't have a `countedDelete` attached. In fact, this comment covers one such scenario for how that's possible -- a counted delete followed by a `put()` will leave the entry in the `PRESENT` state but with a non-null `countedDelete` attached. And the if another `put` came in for the same key, it looks like we'd hit the `PRESENT` case above and fail to carry forward the counted delete?

I also just don't understand the motivation for this code movement. What does it have to do with the rest of the commit?

Reply:
Removed in this commit https://github.com/cloudflare/workerd/pull/1915/commits/534ac7f9a9abab840d9030c1e0b9ca859a03c083